homeassistant/components/modbus/light.py

```diff
@@ -0,0 +1,160 @@
+    CONF_INPUT_TYPE,
+    CONF_STATE_OFF,
+    CONF_STATE_ON,
+    CONF_VERIFY,
+    CONF_WRITE_TYPE,
+    MODBUS_DOMAIN,
+)
+from .modbus import ModbusHub
+
+PARALLEL_UPDATES = 1
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant, config: ConfigType, async_add_entities, discovery_info=None
+):
+    """Read configuration and create Modbus lights."""
+    if discovery_info is None:
+        return
+    lights = []
+    for entry in discovery_info[CONF_LIGHTS]:
+        hub: ModbusHub = hass.data[MODBUS_DOMAIN][discovery_info[CONF_NAME]]
+        lights.append(ModbusLight(hub, entry))
+    async_add_entities(lights)
+
+
+class ModbusLight(BasePlatform, LightEntity, RestoreEntity):
+    """Base class representing a Modbus light."""
+
+    def __init__(self, hub: ModbusHub, config: dict) -> None:
+        """Initialize the light."""
+        config[CONF_INPUT_TYPE] = ""
+        config[CONF_DEVICE_CLASS] = ""
+        super().__init__(hub, config)
+        self._is_on = None
+        if config[CONF_WRITE_TYPE] == CALL_TYPE_COIL:
+            self._write_type = CALL_TYPE_WRITE_COIL
+        else:
+            self._write_type = CALL_TYPE_WRITE_REGISTER
+        self._command_on = config[CONF_COMMAND_ON]
+        self._command_off = config[CONF_COMMAND_OFF]
+        if CONF_VERIFY in config:
+            if config[CONF_VERIFY] is None:
+                config[CONF_VERIFY] = {}
+            self._verify_active = True
+            self._verify_address = config[CONF_VERIFY].get(
+                CONF_ADDRESS, config[CONF_ADDRESS]
+            )
+            self._verify_type = config[CONF_VERIFY].get(
+                CONF_INPUT_TYPE, config[CONF_WRITE_TYPE]
+            )
+            self._state_on = config[CONF_VERIFY].get(CONF_STATE_ON, self._command_on)
+            self._state_off = config[CONF_VERIFY].get(CONF_STATE_OFF, self._command_off)
+        else:
+            self._verify_active = False
+
+    async def async_added_to_hass(self):
+        """Handle entity which will be added."""
+        await self.async_base_added_to_hass()
+        state = await self.async_get_last_state()
+        if state:
+            self._is_on = state.state == STATE_ON
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return self._is_on
+
+    async def async_turn_on(self, **kwargs):
+        """Set switch on."""
+
+        result = await self._hub.async_pymodbus_call(
+            self._slave, self._address, self._command_on, self._write_type
+        )
+        if result is None:
+            self._available = False
+            self.async_write_ha_state()
+        else:
+            self._available = True
+            if self._verify_active:
+                await self.async_update()
+            else:
+                self._is_on = True
+                self.async_write_ha_state()
+
+    async def async_turn_off(self, **kwargs):
+        """Set switch off."""
```

Comment:
```suggestion
        """Set light off."""
```

Reply:
Done (my morning coffee is still in preparation).